### PR TITLE
- add luple/1.2

### PR DESCRIPTION
**Pull request by SSE4**
Specify library name and version:  **luple/1.2**

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/luple/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+class LupleConan(ConanFile):
+    name = "luple"
+    license = "Unlicense"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/alexpolt/luple"
+    description = "Home to luple, nuple, C++ String Interning, Struct Reader and C++ Type Loophole"
+    topics = ("conan", "loophole", "luple", "nuple", "struct", "intern")
+    settings = "compiler"
+
+    def configure(self):
```
Comment (uilianries) [inline recipes/luple/all/conanfile.py:15]:
```suggestion
    def validate(self):
```

File: recipes/luple/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import os
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
```
Comment (uilianries) [inline recipes/luple/all/conanfile.py:4]:
```suggestion

required_conan_version = ">=1.32.0"
```

validate() is 1.32.0

**Review by uilianries — APPROVED** [review]

File: recipes/luple/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import os
+from conans import ConanFile, CMake, tools
```
Comment (SpaceIm) [inline recipes/luple/all/conanfile.py:2]:
```suggestion
from conans import ConanFile, tools
```

File: recipes/luple/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        }
+        compiler = str(self.settings.compiler)
+        if compiler not in minimal_version:
+            self.output.warn(
+                "%s recipe lacks information about the %s compiler standard version support" % (self.name, compiler))
+            self.output.warn(
+                "%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
+            return
+        version = tools.Version(self.settings.compiler.version)
+        if version < minimal_version[compiler]:
```
Comment (SpaceIm) [inline recipes/luple/all/conanfile.py:34]: It will raise if clang >= 3.4 and < 4

Comment (SSE4) [reply recipes/luple/all/conanfile.py:34]: I have taken it from the frozen, any better alternative?

Comment (SpaceIm) [reply recipes/luple/all/conanfile.py:34]: Comment edited, it's fine like this ;)

**Review by SpaceIm — APPROVED** [review]

**Review by czoido — APPROVED** [review]

File: recipes/luple/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
+import os
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.32.0"
+
+class LupleConan(ConanFile):
+    name = "luple"
+    license = "Unlicense"
```
Comment (prince-chrismc) [inline recipes/luple/all/conanfile.py:9]: https://github.com/alexpolt/luple#license ... It "says" public domain 🤔

File: recipes/luple/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+    def package(self):
+        tools.save(os.path.join(self.package_folder, "licenses", "COPYING"), "Public Domain")
+        self.copy("*.h", dst="include", src=self._source_subfolder)
+
+    def package_id(self):
+        self.info.header_only()
+
+    def package_info(self):
+        self.cpp_info.libdirs = []
```
Comment (prince-chrismc) [inline recipes/luple/all/conanfile.py:54]: If we start adding this for every recipe just for the warning.... we need to make it a hook

If the license needs to be changed, i'd suggest removing it

Comment (SSE4) [reply recipes/luple/all/conanfile.py:54]: see https://github.com/conan-io/hooks/pull/281

File: recipes/luple/all/conanfile.py (hunk 5 above)
Comment (SSE4) [reply recipes/luple/all/conanfile.py:9]: `Unilicense` is SDPX identifier for public domain: https://spdx.org/licenses/Unlicense.html

Comment (prince-chrismc) [reply recipes/luple/all/conanfile.py:9]: Thanks! I did not know

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**